### PR TITLE
[qa] Replace with openwisp-utils-qa-checks

### PR DESCRIPTION
**Pull request by waleko**
Replace old qa checks with `openwisp-utils-qa-checks` (related to openwisp/openwisp-utils#63 for this repo)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,13 +16,5 @@ install:
...
-  - |
-    if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
-      # gets commit message of last commit before pull request merge
-      COMMIT_MESSAGE=$(git log $TRAVIS_PULL_REQUEST_SHA --format=%B -n 1)
-      printf "Checking commit message:\n\n"
-      printf "$COMMIT_MESSAGE\n\n"
-      checkcommit --message "$COMMIT_MESSAGE"
-    fi
+  - openwisp-utils-qa-checks --skip-checkmigrations
+  - ./runsphinx-build
```
Comment (atb00ker) [inline .travis.yml]: Add endline! :joy:
PR for qa-checks fails qa checks! :joy:

File: check_line_length.py [hunk 2]
```diff
@@ -1,70 +0,0 @@
-#!/usr/bin/python
```
Comment (atb00ker) [inline check_line_length.py]: @waleko `flake8` works on python files.
`check_line_length.py` is for RST files, I think that's what @ppabcd is pointing at, let me know if I have misunderstood something!

File: .travis.yml (hunk 1 above)
Comment (waleko) [reply .travis.yml]: Lol 😂 Fixed it

File: check_line_length.py (hunk 2 above)
Comment (waleko) [reply check_line_length.py]: I returned `check_line_length.py`

File: check_line_length.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ def check_url(line):
     """
     Check if there is a url present in the given line
     """
-    pattern = 'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\), ]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
+    pattern = 'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\), ]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```
Comment (atb00ker) [inline check_line_length.py]: Can you share why have you made this change? Was it causing a failure?

File: requirements.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
 sphinx
 sphinx_rtd_theme
-openwisp-utils[qa]>=0.2.1
+openwisp-utils[qa]>=0.3.0
```
Comment (atb00ker) [inline requirements.txt]: Please update to `3.1`! :smile:

File: check_line_length.py (hunk 3 above)
Comment (waleko) [reply check_line_length.py]: Flake8 pointed this out. Moreover, there is no `\(` or `\)` character, so although python handles this, it's a good fix imho.

File: setup.cfg [hunk 5]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+exclude = *migrations*,
+	  ./tests/*settings*.py,
+	  ./*setup*.py
```
Comment (nemesisdesign) [inline setup.cfg]: there's no need for this exclude I think

**Pull request merged**